tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -125,69 +119,19 @@ def publish_functions_package(cls, function_params, package):
 
         cls.log.info('Publishing Function application')
 
-        # provision using Kudu Zip-Deploy
-        if not cls.is_consumption_plan(function_params):
-            publish_creds = web_client.web_apps.list_publishing_credentials(
-                function_params.function_app_resource_group_name,
-                function_params.function_app_name).result()
+        publish_creds = web_client.web_apps.list_publishing_credentials(
+            function_params.function_app_resource_group_name,
+            function_params.function_app_name).result()
 
-            if package.wait_for_status(publish_creds):
-                package.publish(publish_creds)
-            else:
-                cls.log.error("Aborted deployment, ensure Application Service is healthy.")
-        # provision using WEBSITE_RUN_FROM_PACKAGE
+        if package.wait_for_status(publish_creds):
+            package.publish(publish_creds)
+
+            is_consumption = cls.is_consumption_plan(function_params)
+            if package.wait_for_remote_build(publish_creds, is_consumption) and is_consumption:
+                cls._sync_function_triggers(function_params)
```

Comment:
is the wait to sync still needed?

Reply:
Unfortunately, it is still needed for consumption :(
Looks like when build is complete, something has to restart function app.

I started the thread to figure out the future plan, I hope we can drop the support and support sync\async runs..

Reply:
there's no sadness 😢 emoji reaction, but it seems appropriate.

Reply:
My bad, removed `sync_triggers` because it can be passed as a parameter to the deployment API